### DISCUSSION TITLE
tips on using lotusdb?

### DISCUSSION THREAD

**Discussion by pkieltyka** [lotusdblabs/lotusdb#29] [Q&A]
hi there -- nice job on the project. I have a few questions as I'm considering using this for a ~500GB dataset. 

1. Does lotusdb have any requirement that all keys must fit in memory?
2. Any atomic / batch write features? the goal is to ensure the db never gets corrupted, but its fine if a write fails and its known by the application
3. What is your recommendation for building inverted indexes on the data set for easily ability to query against columns of a stored row inside a key. Follow-up to that, any suggestions on logical operations to query AND/OR against those indexes?

**Comment by roseduan** [comment]
sorry to answer you so late.
1. no, lotusdb only keeps part of data in memory, memory usage is according to options `MemtableSize` and `MemtableNums`, and we also use `boltdb` to store index info, which may also take some memory.
2. WriteBatch feature is on the way, see our DOING list: https://github.com/flower-corp/lotusdb/projects/1
3. Sorry I know little about inverted indexes, maybe you can find out how ElasticSearch works
